Muelu build error affecting Nalu

## Bug Report
@trilinos/Muelu

Hello @jhux2, please advise on this new build error based on the previously working Trilinos - details below.

[ 97%] Building CXX object packages/muelu/adapters/CMakeFiles/muelu-adapters.dir/ExplicitInstantiation/MueLu_RefMaxwell.cpp.o
In file included from /home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:59,
                 from /home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/build_nightly_release_8.3.0/packages/muelu/adapters/ExplicitInstantiation/MueLu_RefMaxwell.cpp:52:
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:428:9: error: ‘Thyra’ was not declared in this scope
     RCP<Thyra::PreconditionerBase<Scalar> > thyraPrecH_, thyraPrec22_;
         ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:428:9: note: suggested alternative: ‘Tpetra’
     RCP<Thyra::PreconditionerBase<Scalar> > thyraPrecH_, thyraPrec22_;
         ^~~~~
         Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:428:43: error: template argument 1 is invalid
     RCP<Thyra::PreconditionerBase<Scalar> > thyraPrecH_, thyraPrec22_;
                                           ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:462:16: error: ‘Thyra’ was not declared in this scope
     static RCP<Thyra::PreconditionerBase<Scalar> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
                ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:462:16: note: suggested alternative: ‘Tpetra’
     static RCP<Thyra::PreconditionerBase<Scalar> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
                ^~~~~
                Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:462:50: error: template argument 1 is invalid
     static RCP<Thyra::PreconditionerBase<Scalar> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
                                                  ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:468:16: error: ‘Thyra’ was not declared in this scope
     static RCP<Thyra::PreconditionerBase<double> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
                ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:468:16: note: suggested alternative: ‘Tpetra’
     static RCP<Thyra::PreconditionerBase<double> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
                ^~~~~
                Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:468:50: error: template argument 1 is invalid
     static RCP<Thyra::PreconditionerBase<double> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
                                                  ^
In file included from /home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/build_nightly_release_8.3.0/packages/muelu/adapters/ExplicitInstantiation/MueLu_RefMaxwell.cpp:52:
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp: In member function ‘void MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>::applyInverseAdditive(const MultiVector&, MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>::MultiVector&) const’:
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2438:25: error: request for member ‘is_null’ in ‘((const MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>*)this)->MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>::thyraPrecH_’, which is of non-class type ‘const int’
         if (thyraPrecH_.is_null())
                         ^~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:15: error: ‘Thyra’ was not declared in this scope
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
               ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:15: note: suggested alternative: ‘Tpetra’
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
               ^~~~~
               Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:46: error: template argument 1 is invalid
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                              ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:77: error: parse error in template argument list
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:133: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
 alar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^~~~~~~~~~

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:133: note: suggested alternative: ‘MatrixUtils’
 alar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^~~~~~~~~~
                                                                                              MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:150: error: expected primary-expression before ‘,’ token
 raP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:163: error: expected primary-expression before ‘,’ token
 uchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:177: error: expected primary-expression before ‘,’ token
 st_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:182: error: expected primary-expression before ‘>’ token
 st<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:185: error: ‘::toThyraMultiVector’ has not been declared
 Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^~~~~~~~~~~~~~~~~~

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2441:185: note: suggested alternative: ‘TpetraMultiVector’
 Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
                                                                                              ^~~~~~~~~~~~~~~~~~
                                                                                              TpetraMultiVector
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2442:26: error: ISO C++ forbids declaration of ‘type name’ with no type [-fpermissive]
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
                          ^~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2442:52: error: template argument 1 is invalid
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
                                                    ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2442:76: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
                                                                            ^~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2442:76: note: suggested alternative: ‘MatrixUtils’
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
                                                                            ^~~~~~~~~~
                                                                            MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2442:93: error: expected primary-expression before ‘,’ token
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
                                                                                             ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2443:22: error: base operand of ‘->’ is not a pointer
           thyraPrecH_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraP11res, thyraP11x.ptr(), one, zero);
                      ^~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2443:54: error: ‘Thyra’ is not a class, namespace, or enumeration
           thyraPrecH_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraP11res, thyraP11x.ptr(), one, zero);
                                                      ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2443:71: error: invalid type argument of unary ‘*’ (have ‘int’)
           thyraPrecH_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraP11res, thyraP11x.ptr(), one, zero);
                                                                       ^~~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2443:94: error: request for member ‘ptr’ in ‘thyraP11x’, which is of non-class type ‘int’
           thyraPrecH_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraP11res, thyraP11x.ptr(), one, zero);
                                                                                              ^~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2444:48: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
           RCP<MultiVector> thyXpP11x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraP11x, P11x_->getMap()->getComm());
                                                ^~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2444:48: note: suggested alternative: ‘MatrixUtils’
           RCP<MultiVector> thyXpP11x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraP11x, P11x_->getMap()->getComm());
                                                ^~~~~~~~~~
                                                MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2444:65: error: expected primary-expression before ‘,’ token
           RCP<MultiVector> thyXpP11x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraP11x, P11x_->getMap()->getComm());
                                                                 ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2461:26: error: request for member ‘is_null’ in ‘((const MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>*)this)->MueLu::RefMaxwell<Scalar, LocalOrdinal, GlobalOrdinal, Node>::thyraPrec22_’, which is of non-class type ‘const int’
         if (thyraPrec22_.is_null())
                          ^~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:15: error: ‘Thyra’ was not declared in this scope
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
               ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:15: note: suggested alternative: ‘Tpetra’
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
               ^~~~~
               Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:46: error: template argument 1 is invalid
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                              ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:76: error: parse error in template argument list
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:132: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
 calar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^~~~~~~~~~

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:132: note: suggested alternative: ‘MatrixUtils’
 calar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^~~~~~~~~~
                                                                                              MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:149: error: expected primary-expression before ‘,’ token
 yraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:162: error: expected primary-expression before ‘,’ token
 uchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:176: error: expected primary-expression before ‘,’ token
 st_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:181: error: expected primary-expression before ‘>’ token
 st<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:184: error: ‘::toThyraMultiVector’ has not been declared
 Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^~~~~~~~~~~~~~~~~~

/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2464:184: note: suggested alternative: ‘TpetraMultiVector’
 Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
                                                                                              ^~~~~~~~~~~~~~~~~~
                                                                                              TpetraMultiVector
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2465:26: error: ISO C++ forbids declaration of ‘type name’ with no type [-fpermissive]
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
                          ^~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2465:52: error: template argument 1 is invalid
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
                                                    ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2465:75: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
                                                                           ^~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2465:75: note: suggested alternative: ‘MatrixUtils’
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
                                                                           ^~~~~~~~~~
                                                                           MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2465:92: error: expected primary-expression before ‘,’ token
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
                                                                                            ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2466:23: error: base operand of ‘->’ is not a pointer
           thyraPrec22_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraD0res, thyraD0x.ptr(), one, zero);
                       ^~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2466:55: error: ‘Thyra’ is not a class, namespace, or enumeration
           thyraPrec22_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraD0res, thyraD0x.ptr(), one, zero);
                                                       ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2466:72: error: invalid type argument of unary ‘*’ (have ‘int’)
           thyraPrec22_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraD0res, thyraD0x.ptr(), one, zero);
                                                                        ^~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2466:93: error: request for member ‘ptr’ in ‘thyraD0x’, which is of non-class type ‘int’
           thyraPrec22_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraD0res, thyraD0x.ptr(), one, zero);
                                                                                             ^~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2467:47: error: ‘ThyraUtils’ is not a member of ‘Xpetra’
           RCP<MultiVector> thyXpD0x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraD0x, D0x_->getMap()->getComm());
                                               ^~~~~~~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2467:47: note: suggested alternative: ‘MatrixUtils’
           RCP<MultiVector> thyXpD0x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraD0x, D0x_->getMap()->getComm());
                                               ^~~~~~~~~~
                                               MatrixUtils
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2467:64: error: expected primary-expression before ‘,’ token
           RCP<MultiVector> thyXpD0x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraD0x, D0x_->getMap()->getComm());
                                                                ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp: At global scope:
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2953:9: error: ‘Thyra’ was not declared in this scope
     RCP<Thyra::PreconditionerBase<Scalar> > StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
         ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2953:9: note: suggested alternative: ‘Tpetra’
     RCP<Thyra::PreconditionerBase<Scalar> > StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
         ^~~~~
         Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2953:43: error: template argument 1 is invalid
     RCP<Thyra::PreconditionerBase<Scalar> > StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
                                           ^
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2961:9: error: ‘Thyra’ was not declared in this scope
     RCP<Thyra::PreconditionerBase<double> > StratimikosWrapper<double,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
         ^~~~~
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2961:9: note: suggested alternative: ‘Tpetra’
     RCP<Thyra::PreconditionerBase<double> > StratimikosWrapper<double,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
         ^~~~~
         Tpetra
/home/spdomin/gitHubWork/nightlyBuildAndTest/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2961:43: error: template argument 1 is invalid
     RCP<Thyra::PreconditionerBase<double> > StratimikosWrapper<double,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
                                           ^
make[2]: *** [packages/muelu/adapters/CMakeFiles/muelu-adapters.dir/ExplicitInstantiation/MueLu_RefMaxwell.cpp.o] Error 1
make[1]: *** [packages/muelu/adapters/CMakeFiles/muelu-adapters.dir/all] Error 2
make: *** [all] Error 2


### Description

Build error as noted above. Clean SHA1: Trilinos/develop SHA1: 90422f3c55f5bd257613ddbd3a1cb92c66eb4326

### Steps to Reproduce
1. SHA1:  [Trilinos/develop SHA1: 83daac642e071c0db095cd68788f6da7bedef4b3]
1. Configure script:  [https://github.com/NaluCFD/Nalu/blob/master/build/do-configTrilinos_release]

--- a/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp
+++ b/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp
@@ -425,7 +425,9 @@ namespace MueLu {
     Teuchos::RCP<Ifpack2::Preconditioner<Scalar,LocalOrdinal,GlobalOrdinal,Node> > hiptmairPreSmoother_, hiptmairPostSmoother_;
 #endif
     bool useHiptmairSmoothing_;
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
     RCP<Thyra::PreconditionerBase<Scalar> > thyraPrecH_, thyraPrec22_;
+#endif
     //! Various matrices
     Teuchos::RCP<Matrix> SM_Matrix_, D0_Matrix_, D0_T_Matrix_, M0inv_Matrix_, M1_Matrix_, Ms_Matrix_;
     Teuchos::RCP<Matrix> A_nodal_Matrix_, P11_, R11_, AH_, A22_, Addon_Matrix_;
@@ -457,6 +459,8 @@ namespace MueLu {
     mutable Teuchos::RCP<MultiVector> P11res_, P11x_, D0res_, D0x_, residual_, P11resTmp_, P11xTmp_, D0resTmp_, D0xTmp_, D0TR11Tmp_;
   };
 
+
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
   template<typename Scalar,class LocalOrdinal,class GlobalOrdinal,class Node>
   struct StratimikosWrapper {
     static RCP<Thyra::PreconditionerBase<Scalar> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
@@ -468,6 +472,7 @@ namespace MueLu {
     static RCP<Thyra::PreconditionerBase<double> > setupStratimikosPreconditioner(RCP<Xpetra::Matrix<double,LocalOrdinal,GlobalOrdinal,Node> > A,
                                                                                      RCP<ParameterList> params);
   };
+#endif
 
 
 } // namespace

--- a/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp
+++ b/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp
@@ -928,6 +928,7 @@ namespace MueLu {
           params->set("printCommInfo",          true);
           GetOStream(Statistics2) << PerfUtils::PrintMatrixInfo(*AH_, "AH", params);
         }
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
         if (precList11_.isType<std::string>("Preconditioner Type")) {
           // build a Stratimikos preconditioner
           if (precList11_.get<std::string>("Preconditioner Type") == "MueLu") {
@@ -935,7 +936,9 @@ namespace MueLu {
             userParamList.set<RCP<RealValuedMultiVector> >("Coordinates", CoordsH_);
           }
           thyraPrecH_ = StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(AH_, rcp(&precList11_, false));
-        } else {
+        } else
+#endif
+          {
           // build a MueLu hierarchy
 
           if (!reuse) {
@@ -1166,6 +1169,7 @@ namespace MueLu {
           params->set("printCommInfo",          true);
           GetOStream(Statistics2) << PerfUtils::PrintMatrixInfo(*A22_, "A22", params);
         }
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
         if (precList22_.isType<std::string>("Preconditioner Type")) {
           // build a Stratimikos preconditioner
           if (precList22_.get<std::string>("Preconditioner Type") == "MueLu") {
@@ -1173,7 +1177,9 @@ namespace MueLu {
             userParamList.set<RCP<RealValuedMultiVector> >("Coordinates", Coords_);
           }
           thyraPrec22_ = StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(A22_, rcp(&precList22_, false));
-        } else {
+        } else
+#endif
+          {
           // build a MueLu hierarchy
           if (!reuse) {
             ParameterList& userParamList = precList22_.sublist("user data");
@@ -2359,7 +2365,6 @@ namespace MueLu {
   void RefMaxwell<Scalar,LocalOrdinal,GlobalOrdinal,Node>::applyInverseAdditive(const MultiVector& RHS, MultiVector& X) const {
 
     Scalar one = Teuchos::ScalarTraits<Scalar>::one();
-    Scalar zero = Teuchos::ScalarTraits<Scalar>::zero();
 
     { // compute residual
 
@@ -2435,15 +2440,18 @@ namespace MueLu {
         // Replace maps with maps with a subcommunicator
         P11res_->replaceMap(AH_->getRangeMap());
         P11x_  ->replaceMap(AH_->getDomainMap());
-        if (thyraPrecH_.is_null())
-          HierarchyH_->Iterate(*P11res_, *P11x_, numItersH_, true);
-        else {
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
+        if (!thyraPrecH_.is_null()) {
+          Scalar zero = Teuchos::ScalarTraits<Scalar>::zero();
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraP11x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11x_));
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraP11res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(P11res_);
           thyraPrecH_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraP11res, thyraP11x.ptr(), one, zero);
           RCP<MultiVector> thyXpP11x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraP11x, P11x_->getMap()->getComm());
           *P11x_ = *thyXpP11x;
         }
+        else
+#endif
+           HierarchyH_->Iterate(*P11res_, *P11x_, numItersH_, true);
         P11x_  ->replaceMap(origXMap);
         P11res_->replaceMap(origRhsMap);
       }
@@ -2458,15 +2466,18 @@ namespace MueLu {
         // Replace maps with maps with a subcommunicator
         D0res_->replaceMap(A22_->getRangeMap());
         D0x_  ->replaceMap(A22_->getDomainMap());
-        if (thyraPrec22_.is_null())
-          Hierarchy22_->Iterate(*D0res_, *D0x_, numIters22_, true);
-        else {
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
+        if (!thyraPrec22_.is_null()) {
+          Scalar zero = Teuchos::ScalarTraits<Scalar>::zero();
           RCP<Thyra::MultiVectorBase<Scalar> >       thyraD0x   = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<Scalar> >(Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0x_));
           RCP<const Thyra::MultiVectorBase<Scalar> > thyraD0res = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyraMultiVector(D0res_);
           thyraPrec22_->getUnspecifiedPrecOp()->apply(Thyra::NOTRANS, *thyraD0res, thyraD0x.ptr(), one, zero);
           RCP<MultiVector> thyXpD0x = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toXpetra(thyraD0x, D0x_->getMap()->getComm());
           *D0x_ = *thyXpD0x;
         }
+        else
+#endif
+          Hierarchy22_->Iterate(*D0res_, *D0x_, numIters22_, true);
         D0x_  ->replaceMap(origXMap);
         D0res_->replaceMap(origRhsMap);
       }
@@ -2948,6 +2959,7 @@ namespace MueLu {
 
   }
 
+#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
 
   template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
     RCP<Thyra::PreconditionerBase<Scalar> > StratimikosWrapper<Scalar,LocalOrdinal,GlobalOrdinal,Node>::setupStratimikosPreconditioner(RCP<Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> > A,
@@ -2962,8 +2974,6 @@ namespace MueLu {
                                                                                                                                        RCP<ParameterList> params) {
 
     typedef double Scalar;
-
-#if defined(HAVE_MUELU_STRATIMIKOS) && defined(HAVE_MUELU_THYRA)
 
     // Build Thyra linear algebra objects
     RCP<const Thyra::LinearOpBase<Scalar> > thyraA = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyra(Teuchos::rcp_dynamic_cast<Xpetra::CrsMatrixWrap<Scalar,LocalOrdinal,GlobalOrdinal,Node>>(A)->getCrsMatrix());
@@ -2988,12 +2998,8 @@ namespace MueLu {
     precFactory->initializePrec(Thyra::defaultLinearOpSource(thyraA), prec.get(), Thyra::SUPPORT_SOLVE_UNSPECIFIED);
 
     return prec;
-#else
-    TEUCHOS_TEST_FOR_EXCEPTION(true, std::runtime_error,
-                               "Need Thyra & Stratimikos support");
-#endif
   }
-
+#endif
 
 } // namespace
 

